File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
 let total = brown + yellow;
 let result = [];
  for(let i = 3; i < total; i++) {
    if(total % i == 0) {
        let hori = total / i;
        if(hori <= i) {
            if((hori - 2) * (i - 2) == yellow) {
               result[0] = i;
               result[1] = total / i;
               break;  
            }
        }
    }
  }
    return result;
}

File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int total = brown + yellow;
        int[] result = new int[2];
        for(int i = 3; i < total; i++) {
            if(total % i == 0){
                int vertical = total / i;
                if(vertical <= i) {
                   if((vertical - 2) * (i - 2) == yellow) {
                       result[0] = i;
                       result[1] = vertical;
                   } 
                }
            }
        }
        return result;
    }
}